cmd2/cmd2.py

```diff
@@ -2962,8 +2962,8 @@ def do_set(self, args: argparse.Namespace) -> None:
         # See if we need to call a change hook for this settable
         if orig_value != new_value:
             onchange_hook = getattr(self, '_onchange_{}'.format(param), None)
-            if onchange_hook is not None:
-                onchange_hook(old=orig_value, new=new_value)
+            if onchange_hook is not None and callable(onchange_hook):
```

Comment:
Your comments made me change my mind - if a developer uses this incorrectly it would be better to throw an exception than fail silently.  Sorry 😐

Reply:
No worries. I've reverted the change, and am going to merge this PR.